tests/io_components/test_read.py

```diff
@@ -95,8 +94,7 @@ def test_dispatch_metapartitions_query_partition_on(
 @pytest.mark.parametrize(
     "predicates",
     [
-        [[("P", "==", 2)], [("TARGET", "==", 500)]],
-        [[("P", "in", [2])], [("TARGET", "in", [500])]],
+        [[("L", "in", [2])], [("TARGET", "in", [500])]],
```

Comment:
Please review this thoroughly. It seems to me that we had a bug in the partition pruning before this refactoring. The previously tested behaviour seems wrong to me.